### PR TITLE
Starting a refactoring around RunContext and Docker local/remote Api

### PR DESCRIPTION
**Pull request by dgageot**
My goals are:
+ Make the code safer
+ Make is easier to mock
+ Reduce the number of parameters that are to be passed along such as insecureregistries

This is just the beginning of the refactoring but already improves a bit the codebase

### CODE REVIEW

File: pkg/skaffold/runner/runner.go [hunk 1]
```diff
@@ -197,11 +191,13 @@ func getDeployer(runCtx *runcontext.RunContext) (deploy.Deployer, error) {
 	}
 }
 
-func getTagger(t latest.TagPolicy, customTag string) (tag.Tagger, error) {
+func getTagger(runCtx *runcontext.RunContext) (tag.Tagger, error) {
```
Comment (nkubala) [inline pkg/skaffold/runner/runner.go]: so is the end goal here just to make everything consistent? maybe I'm mistaken but I thought that you wanted to move away from the runcontext in these constructors. just trying to understand

File: pkg/skaffold/docker/client.go [hunk 2]
```diff
@@ -40,23 +40,22 @@ import (
...
 var (
 	dockerAPIClientOnce sync.Once
 	dockerAPIClient     LocalDaemon
 	dockerAPIClientErr  error
 )
 
-// NewAPIClient guesses the docker client to use based on current kubernetes context.
-func NewAPIClient(forceRemove bool, insecureRegistries map[string]bool) (LocalDaemon, error) {
+// NewAPIClientImpl guesses the docker client to use based on current kubernetes context.
+func NewAPIClientImpl(runCtx *runcontext.RunContext) (LocalDaemon, error) {
```
[the remark this answers is not in the document]
Comment (dgageot) [reply pkg/skaffold/docker/client.go]: Yeah, I could try that

File: pkg/skaffold/runner/runner.go (hunk 1 above)
Comment (dgageot) [reply pkg/skaffold/runner/runner.go]: :-) For now, I'm aiming at consistency.

**Comment by dgageot** [comment]
@nkubala could you take another look? That would be awesome because this one is a bit painful to rebase.

**Pull request merged**